### PR TITLE
Updates custom menu borders and item styles

### PR DESCRIPTION
**Pull request by daviddossett**
Continues work from #149187 to modernize custom menu styles by...
- Updating menu items with margin and border radius (like W11/macOS/new quick pick list styles)
- Adding subtle border to default dark and light themes to help with contrast against background
- Updates background color on dark theme—the one I originally used to update feels washed out

## Examples

CleanShot 2022-10-20 at 11 35 05@2x

CleanShot 2022-10-20 at 11 35 23@2x

CleanShot 2022-10-20 at 11 34 35@2x

CleanShot 2022-10-20 at 11 34 54@2x

### CODE REVIEW

File: src/vs/base/browser/ui/menu/menu.ts [hunk 1]
```diff
@@ -1251,7 +1253,7 @@ ${formatRule(Codicon.menuSubmenu)}
 /* Vertical Action Bar Styles */
 
 .monaco-menu .monaco-action-bar.vertical {
-	padding: .6em 0;
```
Comment (daviddossett) [inline src/vs/base/browser/ui/menu/menu.ts:1254]: @sbatten does this affect more than just the menus? The classname is suspicious.

Comment (sbatten) [reply src/vs/base/browser/ui/menu/menu.ts:1254]: the only action bars that are vertical are menus and the activity bar icons I'm pretty sure. but I'm guessing this is overridden by activity bar styles.

Comment (sbatten) [reply src/vs/base/browser/ui/menu/menu.ts:1254]: oh also, since its under .monaco-menu, only menu widgets are affected

Comment (daviddossett) [reply src/vs/base/browser/ui/menu/menu.ts:1254]: Ok cool, I'll verify that the change doesn't affect anything else. Let me know if there are any other concerns 👍

**Review by sbatten — APPROVED** [review]

**Pull request merged**